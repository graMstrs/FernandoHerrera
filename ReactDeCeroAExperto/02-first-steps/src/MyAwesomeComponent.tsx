import { useState, type CSSProperties } from "react";
export function MyAwesomeComponent() {

    const [isActive, setIsActive] = useState(false)
    const firstName = 'Gerardo';
    const lastName = 'Tavera';

    const favoriteGames = ['Elden Ring', 'Zelda', 'Doom'];

    const toggleIsActive = () => {
        setIsActive (!isActive)
    }

    const address = {
        zipCode: 'ABC-123',
        street: 'Main Stree 123',
        country: 'Canada'
    }

    const addressStyle: CSSProperties = {
        backgroundColor: isActive ? 'green' : 'blue', 
        borderRadius: '10px',
        padding: 10, 
        marginTop: 30,
    }

    return (
        <div> 
            <h1 data-testid="first-name-title"> {firstName} </h1>
            <h3 > {lastName} </h3>

            <p className="mi-clase-favorita"> {favoriteGames.join(', ')} </p>
            <p> {2 + 2 }</p>

            <h1>{isActive ? 'Active' : 'Inactive'}</h1>
            <button style={{  height: '30px', width: '100px' }} 
                    onClick={ toggleIsActive}> 
                    { isActive? 'Desactivar' : 'Activar' } 
            </button>

            <p style={ addressStyle }>
                { JSON.stringify(address) }
            </p>
        </div>
    )
}