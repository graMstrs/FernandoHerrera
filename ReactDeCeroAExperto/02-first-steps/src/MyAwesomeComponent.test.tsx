import { describe, test, expect } from "vitest";
import { render, screen } from '@testing-library/react';

import { MyAwesomeComponent } from "./MyAwesomeComponent";

describe ('MyAwesomeComponent', () => {

    test('should render first name and last name', () => {
        const {container} = render (<MyAwesomeComponent />);
        //screen.debug();
        //console.log(container.innerHTML);

        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');
        
        expect(h1?.innerHTML).toContain('Gerardo');
        expect(h3?.innerHTML).toContain('Tavera');
    }),

    test('should render first name and last name - screen', () => {
        render (<MyAwesomeComponent />);
        //screen.debug();
        //console.log(container.innerHTML);

        // const h1 = screen.getByRole('heading', {
        //     level: 1,
        // })

    }),
    
    test('should match snapshot', () => {
        const { container } = render(<MyAwesomeComponent />);
        
        expect(container).toMatchSnapshot();
    });


});