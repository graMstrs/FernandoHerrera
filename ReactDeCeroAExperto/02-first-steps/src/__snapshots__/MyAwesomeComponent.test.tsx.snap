// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeComponent > should match snapshot 1`] = `
<div>
  <div>
    <h1>
       
      Gerardo
       
    </h1>
    <h3>
       
      Tavera
       
    </h3>
    <p>
       
      Elden Ring, Zelda, Doom
       
    </p>
    <p>
       
      4
    </p>
    <h1>
      Inactive
    </h1>
    <button
      style="height: 30px; width: 100px;"
    >
      Activar
    </button>
    <p
      style="background-color: blue; border-radius: 10px; padding: 10px; margin-top: 30px;"
    >
      {"zipCode":"ABC-123","street":"Main Stree 123","country":"Canada"}
    </p>
  </div>
</div>
`;
